File: java/src/main/java/com/alxarafe/app/application/catalogue/CreateUom.java
package com.alxarafe.app.application.catalogue;

import com.alxarafe.app.domain.catalogue.valueobject.Uom;
import com.alxarafe.app.domain.catalogue.valueobject.UomCode;
import com.alxarafe.app.domain.catalogue.valueobject.UomId;

public final class CreateUom {
    private final UomRepository repository;

    public CreateUom(UomRepository repository) {
        this.repository = repository;
    }

    public Uom execute(String code, String description) {
        if (description.length() > 255) {
            throw new IllegalArgumentException("Uom description cannot exceed 255 characters.");
        }

        Uom uom = new Uom(UomId.generate(), new UomCode(code), description);
        if (repository.findByCode(uom.code()).isPresent()) {
            throw new UomConflict("Uom code already exists.");
        }

        repository.save(uom);
        return uom;
    }
}

File: php/src/Application/Catalogue/CreateUom.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

use Alxarafe\App\Domain\Catalogue\ValueObject\Uom;
use Alxarafe\App\Domain\Catalogue\ValueObject\UomCode;
use Alxarafe\App\Domain\Catalogue\ValueObject\UomId;
use InvalidArgumentException;

final readonly class CreateUom
{
    public function __construct(private UomRepository $repository)
    {
    }

    public function execute(string $code, string $description): Uom
    {
        if (strlen($description) > 255) {
            throw new InvalidArgumentException('Uom description cannot exceed 255 characters.');
        }

        $uom = new Uom(UomId::generate(), new UomCode($code), $description);
        if ($this->repository->findByCode($uom->code()) !== null) {
            throw new UomConflict('Uom code already exists.');
        }

        $this->repository->save($uom);
        return $uom;
    }
}
